pkg/controller/.import-restrictions

```diff
@@ -261,6 +261,8 @@
         "k8s.io/kubernetes/pkg/capabilities",
         "k8s.io/kubernetes/pkg/master/ports",
         "k8s.io/kubernetes/pkg/scheduler/api",
+        "k8s.io/kubernetes/pkg/scheduler/internal",
+        "k8s.io/kubernetes/pkg/scheduler/internal/errchannel",
```

Comment:
we don't want controllers depending on internal scheduler packages, right?

Reply:
Eventually we dont. I plan to remove this coupling in the coming weeks, the code can be refactored to remove this dependency. This is added for now for the scheduler cleanup. Can be marked as a TODO.

Reply:
if this is intended to be a cleanup PR, it would be preferable not to introduce new debt in it. can that dependency be dropped before we mark this internal?

Reply:
@hprateek43, how do you plan to remove the dependency?

Other than that, I don't understand why import-restrictions apply to indirect imports. I feel like it shouldn't.

Reply:
the idea is to first eliminate the utils package and then refactor predicates to remove dependencies on internal packages. But @alculquicondor is also right in his place that transient dependencies should not be considered.

Reply:
Just as a reminder, `.import-restrictions` is a home-grown k8s utility that requires us to whitelist all transitive dependencies. The go compiler prevents `pkg/controller` from directly importing anything in `pkg/scheduler/internal/...`

Reply:
@liggitt This point could be considered. Although now I am planning to remove the error channel altogether

Reply:
> @liggitt This point could be considered

that's a good point, and would remove my objection

> Although now I am planning to remove the error channel altogether

that sounds even better :)